src/DynamoCoreWpf/Controls/NodeAutoCompleteSearchControl.xaml.cs

```diff
@@ -0,0 +1,258 @@
+                RequestShowNodeAutoCompleteSearch(flags);
+            }
+        }
+
+        private void OnSearchTextBoxTextChanged(object sender, TextChangedEventArgs e)
+        {
+            BindingExpression binding = ((TextBox)sender).GetBindingExpression(TextBox.TextProperty);
+            if (binding != null)
+                binding.UpdateSource();
+
+            if (ViewModel != null)
+                ViewModel.SearchCommand.Execute(null);
+        }
+
+        private void OnMouseLeftButtonUp(object sender, MouseButtonEventArgs e)
+        {
+            var listBoxItem = sender as ListBoxItem;
+            if (listBoxItem == null || e.OriginalSource is Thumb) return;
+
+            ExecuteSearchElement(listBoxItem);
+            OnRequestShowNodeAutoCompleteSearch(ShowHideFlags.Hide);
+            e.Handled = true;
+        }
+
+        private void ExecuteSearchElement(ListBoxItem listBoxItem)
+        {
+            var searchElement = listBoxItem.DataContext as NodeSearchElementViewModel;
+            if (searchElement != null)
+            {
+                searchElement.Position = ViewModel.InCanvasSearchPosition;
+                PortViewModel port = ViewModel.PortViewModel;
+                searchElement.CreateAndConnectCommand.Execute(port.PortModel);
+            }
+        }
+
+        private void OnMouseEnter(object sender, MouseEventArgs e)
+        {
+            FrameworkElement fromSender = sender as FrameworkElement;
+            if (fromSender == null) return;
+
+            toolTipPopup.DataContext = fromSender.DataContext;
+            toolTipPopup.IsOpen = true;
+        }
+
+        private void OnMouseLeave(object sender, MouseEventArgs e)
+        {
+            toolTipPopup.DataContext = null;
+            toolTipPopup.IsOpen = false;
+        }
+
+        private void OnNodeAutoCompleteSearchControlVisibilityChanged(object sender, DependencyPropertyChangedEventArgs e)
+        {
+            // If visibility  is false, then stop processing it.
+            if (!(bool)e.NewValue)
+                return;
+
+            // When launching this control, always start with clear search term.
+            SearchTextBox.Clear();
+
+            // Visibility of textbox changed, but text box has not been initialized(rendered) yet.
+            // Call asynchronously focus, when textbox will be ready.
+            Dispatcher.BeginInvoke(new Action(() =>
+            {
+                SearchTextBox.Focus();
+                //ViewModel.InitializeDefaultAutoCompleteCandidates();
+                ViewModel.PopulateAutoCompleteCandidates();
+            }), DispatcherPriority.Loaded);
+        }
+
+        private void OnMembersListBoxUpdated(object sender, DataTransferEventArgs e)
+        {
+            var membersListBox = sender as ListBox;
+            // As soon as listbox renders, select first member.
+            membersListBox.ItemContainerGenerator.StatusChanged += OnMembersListBoxIcgStatusChanged;
```

Comment:
What does the name mean - Icg status?

Reply:
This is from the legacy code for the control we are mimicing, it means `ItemContainerGenerator`